5_evaluate.R

```diff
@@ -0,0 +1,143 @@
+source('5_evaluate/src/eval_utility_fxns.R')
+source('4_visualize/src/plot_data_utility_fxns.R')
+
+p5 <- list(
+  ##### Evaluate GCM model output #####
+  
+  ##### Evaluate NLDAS model output #####
+  
+  ### Prep predictions and observations
+  
+  # Get NLDAS preds in long format
+  # As-is, model results are read in by `write_glm_output()` in the `3_extract` step
+  # We could modify that function and split the `p3_nldas_glm_uncalibrated_output_feathers`
+  # target into two - one that reads in the output in wide format - `p3_nldas_glm_uncalibrated_output`, 
+  # and a second that writes it to a feather file - `p3_nldas_glm_uncalibrated_output_feathers`
+  # then we could use that first target here to get NLDAS in long format, per https://github.com/USGS-R/mntoha-data-release/blob/main/src/eval_utils.R#L23-L30
+  tar_target(
+    p5_nldas_preds_long,
+    munge_long(p3_nldas_glm_uncalibrated_output) %>%
+      mutate(site_id = p2_nldas_glm_uncalibrated_run_groups$site_id, .before=1) %>%
+      select(-ice),
+    pattern = map(p3_nldas_glm_uncalibrated_output, p2_nldas_glm_uncalibrated_run_groups)
+  ),
+  
+  # load site observations
+  # Bring in `7b_temp_merge/out/merged_temp_data_daily.feather` from `lake-temperature-model-prep`
+  # (would be target in 1_prep.R)
+  tar_target(p5_obs),
+  
+  # Filter obs to sites for which we have NLDAS output
+  # Use https://github.com/USGS-R/mntoha-data-release/blob/main/src/file_utils.R#L441-L446
+  # Passing `driver_start_date` and `driver_end_date` from `p1_nldas_dates` 
+  tar_target(p5_filtered_obs),
+  
+  # Subset obs to those for sites w/ >= 10 dates with observations
+  # Use https://github.com/USGS-R/mntoha-data-release/blob/main/src/eval_utils.R#L4-L11
+  tar_target(p5_obs_for_eval,
+             filter_min_dates(p5_filtered_obs, min_dates = 10)),
+  
+  # Get vector of evaluation sites
+  tar_target(p5_eval_sites,
+             p5_obs_for_eval %>% pull(site_id) %>% unique()),
+  
+  # Group subsetted obs by site, set up tar_group()
+  tar_target(p5_obs_for_eval_groups,
+             p5_obs_for_eval %>%
+               group_by(site_id) %>%
+               tar_group(),
+             iteration = "group"),
+  
+  # filter long-format NLDAS preds to only evaluation sites
+  tar_target(
+    p5_nldas_preds_eval,
+    p5_nldas_preds_long %>%
+      filter(site_id %in% p5_eval_sites)
+  ),
```

Comment:
The more targets we have, the more checking that `targets` does to determine whether something needs to rebuild or not, which results in lengthier builds. Unless you need to reuse some of these targets as inputs to later ones or a particular step is expensive to rebuild, I would suggest collapsing some of these inside of a function that preps/filters.

Reply:
Yes I think that makes sense